dashboard/app/helpers/users_helper.rb

```diff
@@ -1,2 +1,59 @@
 module UsersHelper
+
+  # Summarize the current user's progress within a certain script.
+  def summarize_user_progress(script, callouts_seen = nil, videos_seen = nil)
+    if (!current_user)
+      lines = session[:lines] || 0
+      user_data = {
+        linesOfCode: lines,
+        linesOfCodeText: t('nav.popup.lines', lines: lines),
```

Comment:
I'm going to remove `videosSeen` and `calloutsSeen` because it's part of a future change, and none of the callers of `summarize_user_progress` today pass in more than one argument.